shared_models/shared_models.py

```diff
@@ -100,13 +103,54 @@ class Endorsement(BaseModel):
     state: str
 
 
+class ServiceDecorator(TypedDict):
+    endpoint: str
+    recipient_keys: List[str]
+    routing_keys: Optional[List[str]]
+
+
+class Oob(BaseModel):
+    role: Optional[str] = str
+    invi_msg_id: Optional[str] = None
+    state: Optional[str] = str
+    trace: Optional[str] = None
+    our_recipient_key: Optional[str] = None
+    oob_id: Optional[str] = None
+    invitation: Union[InvitationMessage, Mapping[str, Any]] = None
+    created_at: Optional[str] = None
+    connection_id: Optional[str] = None
+    updated_at: Optional[str] = None
+    attach_thread_id: Optional[str] = None
+    multi_use: Optional[bool] = False
+    our_recipient_key: Optional[str] = None
+    their_service: Optional[Union[ServiceDecorator, Any]] = None
+    our_service: Optional[Union[ServiceDecorator, Any]] = None
+
```

Comment:
Are these important to expose? We can always keep the model ligth for now and add properties as needed. The smaller the api / number of properties, the easier it will be for us to change it later.

Reply:
Agreed. I don't think we do either. Just kept all of the ones from aca-py